ext/standard/array.c

```diff
@@ -5162,12 +5150,12 @@ static void php_array_diff_key(INTERNAL_FUNCTION_PARAMETERS, int data_compare_ty
 			}
 			if (ok) {
 				Z_TRY_ADDREF_P(val);
-				zend_hash_index_update(Z_ARRVAL_P(return_value), p->h, val);
+				zend_hash_index_update(Z_ARRVAL_P(return_value), h, val);
```

Comment:
Same question here.

Reply:
I had originally changed these to zend_hash_index_add_new, but reverted that in the process of refactoring the call to use the macro ZEND_HASH_FOREACH_KEY_VAL_IND and debugging test failures. Fixed.